### ISSUE TITLE
Relative refs in JSON schemas do not work in a particular case

### ISSUE THREAD

**Issue by ddenisenko** [raml-org/raml-js-parser-2#583]
Moving JSON-related  comment from https://github.com/raml-org/raml-js-parser-2/issues/580 here:

This also affects JSON schemas - the circumstances under which this occurs are:

the RAML file uses !include to pull in a schema from a subdirectory of where the RAML file itself resides
the schema uses $ref to include a definition from another schema file using a path relative to the location of the including schema
See also raml2html/raml2html#294

**Comment by ddenisenko** [comment]
So, here is the example I constructed in order to reproduce the issue.

Main RAML file:
```
#%RAML 1.0
title: API

types:
  TestSchema: !include schemas/complex.json


/resource:
  post:
    body:
      application/json:
        type: TestSchema
        example: !include examples/complex.json
```

schemas/complex.json file:
```
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "simple": {"$ref": "simple.json"},
    "name": {"type": "string"}
  },
  "required": ["simple","name"]
}
```

schemas/simple.json file:
```
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "string"
}
```

examples/complex.json file:
```
{
  "simple": "test",
  "name": "Test"
}
```

This example works for me, notice that both `complex.json` and `simple.json` are located in the same folder, and that folder is different from the one, main RAML file is located in.

It looks like in order to reproduce the issue you have at hands we'll need your RAML and JSON files attached including the folder structure.

**Comment by edjannoo** [comment]
Thanks for looking at this. It appears that this issue only presents itself when the schema is pulled in with `!include` at the point of use rather than if it is used to declare a named type within the RAML file. 

So, taking your example but changing the RAML to the following triggers the error:

```
#%RAML 1.0
title: API

/resource:
  post:
    body:
      application/json:
        type: !include schemas/complex.json
        example: !include examples/complex.json
```

With your example, the api-workbench plugin for Atom (which I understand uses raml-js-parser) does not give any errors, but with my version it flags up an error that the example does not conform to schema.

This is the opposite problem to one that exists with raml-java-parser (raml-org/raml-java-parser#310), which does successfully resolve `$ref` in schemas which are pulled in at the point of use but does not when they are declared as a named type.

**Comment by ddenisenko** [comment]
Reproduced, thank you.

**Comment by ddenisenko** [comment]
Fixed. Fix will be released in 1.1.13.
